src/main/java/org/opensearch/knn/index/query/parser/KNNXParserUtil.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.knn.index.query.parser;
+
+import org.opensearch.core.common.ParsingException;
+import org.opensearch.core.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.opensearch.core.xcontent.XContentParserUtils.parseFieldsValue;
+import static org.opensearch.knn.index.query.KNNQueryBuilder.NAME;
+
+public final class KNNXParserUtil {
+
+    public static Map<String, Object> parseJsonObject(XContentParser parser) throws IOException {
```

Comment:
Do we need this if XContentParser has map() method? https://github.com/opensearch-project/OpenSearch/blob/main/libs/core/src/main/java/org/opensearch/core/xcontent/XContentParser.java#L166

Reply:
Wasn't aware of the map, it pretty much does the same thing except the initial start object check. I will switch to that and delete this file

Reply:
Making this a part of another PR